infrastructure/monitoring/kube-prometheus-stack/thanos/README.md

```diff
@@ -0,0 +1,176 @@
+```
+
+Based on this storage class, an object client can now request a bucket by creating an Object Bucket Claim
+
+
+```sh
+$ kubectl apply -f ./manifests/objectBucketClaim.yaml
+```
+
+Now that the object store is configured and a bucket created, it's possible to consume the object storage from an S3 client. In order to access the object storage these three environment variable need to be set.
+
+```sh
+export AWS_HOST=$(kubectl -n default get cm ceph-thanos-bucket -o yaml | grep " BUCKET_HOST" | awk '{print $2}')
+export AWS_ACCESS_KEY_ID=$(kubectl -n default get secret ceph-thanos-bucket -o yaml | grep " AWS_ACCESS_KEY_ID" | awk '{print $2}' | base64 --decode)
+export AWS_SECRET_ACCESS_KEY=$(kubectl -n default get secret ceph-thanos-bucket -o yaml | grep " AWS_SECRET_ACCESS_KEY" | awk '{print $2}' | base64 --decode) 
+```
+
+Once we have exported these values we need to create a configuration file for kubernetes in order make Thanos aware of how to connect to the object storage (replace with the corresponding values)
+
+
+```yaml
+type: s3
+config:
+  bucket: ceph-thanos-bkt
+  endpoint: <<BUCKET_HOST>>
+  access_key: <<AWS_ACCESS_KEY_ID>>
+  secret_key: <<AWS_SECRET_ACCESS_KEY>>
+```
+
+Save it and then create a secret with it using the command
+
+```sh
+$ kubectl -n monitoring create secret generic thanos-objstore-config --from-file=thanos.yaml=./<<your-file-name>>.yaml
+```
+
+And finally the last thing to do is to add the Thanos sidecar to Prometheus with the secret we just created. To do so modify the following lines to the [kube-prometheus-stack-values.yaml](../kube-prometheus-stack-values.yaml#L2647).
+
+
+
+```yaml
+prometheus:
+  ...
+  prometheusSpec:
+    ## Thanos configuration allows configuring various aspects of a Prometheus server in a Thanos environment.
+    ## This section is experimental, it may change significantly without deprecation notice in any release.
+    ## This is experimental and may change significantly without backward compatibility in any release.
+    ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#thanosspec
+    ##
+    thanos: 
+      objectStorageConfig:
+        key: thanos.yaml
+        name: thanos-objstore-config
+      image: quay.io/thanos/thanos:v0.24.0
+      version: v0.24.0
+  ...    
+```
+Moreover, there are these two components to be activated:
+- [thanosService](../kube-prometheus-stack-values.yaml#L1863).
+
+- [thanosServiceMonitor](../kube-prometheus-stack-values.yaml#L1893).
+
+The [thanos-values.yaml](./thanos-values.yaml) defines the Thanos configuration. You must configure the values file according to your specifications; here, we modified it to suit the CrownLabs needs. The main changes are below, the original values file is located [here](https://github.com/bitnami/charts/blob/master/bitnami/thanos/values.yaml).
+
+We have enabled the following components:
+- [query](./thanos-values.yaml#L96)
```

Comment:
Would add a few words to describe what each component does.

Reply:
Merge **Extensions** with **Componets**